proxies/nodejs/handlers/location.ts

```diff
@@ -0,0 +1,146 @@
+import Koa from 'koa'
+import { getEntityFromType, Data } from '../entityTypes'
+
+//HTTP request comes in as string
+type LocationRequestBody = {
+    command: string
+    params: string //[{value: string | number | boolean} | {location: string} | {callbackUrl: string}] 
+    isAsync: boolean
+}
+
+export const handleLocation = async (
+    ctx: Koa.ParameterizedContext,
+    data: Data,
+    body: LocationRequestBody,
+    entity: any
+) => {
+
+    try {
+        const command = body.command
+        const params: string | boolean | number | object = parseParams(
+            JSON.parse(body.params),
+            data
+        )
+
+        // TODO: handle async commands
+        // TODO: handle callback before invoking command
+        const resultData = await invokeCommand(
+            entity,
+            command,
+            params,
+            body.isAsync
+        )
+
+        const entityType = getEntityFromType(resultData.constructor.name)
+
+        const commandId = data.commandResults[entityType.model] !== undefined ?
+            Object.keys(data.commandResults[entityType.model]).length :
+            0
+
+        if (data.commandResults[entityType.model] === undefined) {
+            data.commandResults[entityType.model] = {}
+        }
+        data.commandResults[entityType.model][commandId] = resultData
+
+        ctx.status = 200
+        ctx.set('Location', `command/${entityType.model}/${commandId}`)
+        ctx.body = {
+            entityType: entityType.type,
+            data: resultData,
+            logs: [], // TODO add logs here
+        }
+        console.log('dataObject: ', data)
+        console.log('logger: ', entity.logger.info)
+
+    } catch (error) {
+        console.error(error)
+        if (body.isAsync) {
+            ctx.status = 200
+            ctx.body = {
+                asyncError: error.message,
+                errorCode: error.code,
+            }
+        } else {
+            ctx.status = 200
+            ctx.body = {
+                errorCode: error.code,
+                exception: error.message,
+            }
+
+        }
+    }
+}
+
+const getEntityFromLocation = (location, data) => {
```

Comment:
we don't have typescript linting yet, but could you add the types to the functions params?

Reply:
added to all missing ones, hopefully they seem alright